src/test/kotlin/ar/edu/unahur/obj2/caralibro/UsuarioTest.kt

```diff
@@ -20,15 +117,125 @@ class UsuarioTest : DescribeSpec({
+        pepe.cantidadDeAmigos().shouldBe(0)
+      }
+      it(" Cual es mas amistoso pepe o juana : Es Juana") {
+        juana.esMasAmistosoQue(pepe).shouldBeTrue()
+      }
+      pepe.agregarAmigo(luchoRobles)
+      pepe.agregarAmigo(luchoRobles)
+      pepe.agregarAmigo(luchoRobles)
+      pepe.agregarAmigo(luchoRobles)
+
+      it("Cual es mas amistoso pepe o juana : Es Pepe") {
+        pepe.esMasAmistosoQue(juana).shouldBeTrue()
+      }
+      describe("un me gusta por usuario") {
+        it("la cantidad de me gusta debe ser 3 para fotoEnCuzco y 1 para videoDeCumpleanios") {
+          fotoEnCuzco.cuantasVecesFueVotada().shouldBe(3)
+          videoDeCumpleanios.cuantasVecesFueVotada().shouldBe(1)
+        }
+      }
+      describe("me gusta dados por usuarios que ya lo hicieron no modifican el total de votos de la publicacion") {
+        ferAsam.darMeGusta(fotoEnCuzco)
+        pepe.darMeGusta(videoDeCumpleanios)
+
+        it("la cantidad de me gusta debe ser 3 para fotoEnCuzco y 1 para videoDeCumpleanios") {
+          fotoEnCuzco.cuantasVecesFueVotada().shouldBe(3)
+          videoDeCumpleanios.cuantasVecesFueVotada().shouldBe(1)
+        }
+      }
+      describe("Req 4: Saber si un usuario puede ver una publicacion de otro") {
+        juana.agregarAmigo(pepe)
+
+        it("el usuario siempre puede ver sus publicaciones sin importar el permiso") {
+          juana.puedeVerPublicacion(fotoEnCuzco).shouldBeTrue()
+          juana.puedeVerPublicacion(videoDeCumpleanios).shouldBeTrue()
+          juana.puedeVerPublicacion(saludoCumpleanios).shouldBeTrue()
+        }
+        it("pepe puede ver fotoEnCuzco con permiso soloAmigos") {
+          pepe.puedeVerPublicacion(fotoEnCuzco).shouldBeTrue()
+        }
+        it("pepe no puede ver videoCumpleanios con permiso incluyente") {
+          pepe.puedeVerPublicacion(videoDeCumpleanios).shouldBeFalse()
+        }
+        it("pepe no puede ver saludoCumpleanios con permiso excluyente") {
+          pepe.puedeVerPublicacion(saludoCumpleanios).shouldBeFalse()
+        }
+      }
+      describe("Cambio de permiso fotoEnCuzco a Excluyente") {
+        fotoEnCuzco.cargarPermiso(Excluyente(mutableListOf(pepe, ferAsam, faloiFede)))
+
+        it("pepe no puede ver fotoEnCuzco con permiso excluyente") {
+          pepe.puedeVerPublicacion(fotoEnCuzco).shouldBeFalse()
+        }
+      }
+      describe("Req 5: Mejores amigos de un usuario") {
+        juana.agregarAmigo(jorge)
+        juana.agregarAmigo(diego)
+        juana.agregarAmigo(marcos)
+        juana.agregarPublicacionYPermiso(saludoVacaciones, Excluyente(mutableListOf(marcos, diego)))
+        juana.agregarPublicacionYPermiso(fotoEnArequipa, SoloAmigos(juana.amigos))
+        val listaMejoresAmigos = mutableListOf(ferAsam, luchoRobles, faloiFede)
+
+        it("mejores Amigos: ferAsam, luchoRobles, faloiFede") {
+          juana.mejoresAmigos().shouldContainExactlyInAnyOrder(listaMejoresAmigos)
```

Comment:
Solo para ahorrarse de crear la lista, también se puede hacer así:

```suggestion
          juana.mejoresAmigos().shouldContainExactlyInAnyOrder(ferAsam, luchoRobles, faloiFede)
```